File: test7C#/test7/Model/Burger/HamBurger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace test7.Model.Burger
{
    class HamBurger : SimpleBurger
    {
        public HamBurger() : base("Ham Burger", Convert.ToDecimal(3.99))
        {
            base.Filling(base.Filling() + ",NotHam!");
        }
    }
}

File: test7Go/oop/models/burger/HamBurger.go
package burger

import (
	"fmt"
)

type HamBurger struct {
	SimpleBurger
}

func NewHamBurger() HamBurger {
	simpleBurger := NewSimpleBurger("Ham Burger", 3.99)
	simpleBurger.Filling = fmt.Sprintf("%s,NotHam!", simpleBurger.Filling)

	return HamBurger{simpleBurger}
}
